pkg/cmd/bootstrap.go

```diff
@@ -171,23 +181,30 @@ func initiateInteractiveMode(io *BootstrapParameters, client *utility.Client) er
 	}
 	io.GitOpsWebhookSecret = ui.EnterGitWebhookSecret()
 	io.ServiceRepoURL = utility.AddGitSuffixIfNecessary(ui.EnterServiceRepoURL())
-	if ui.IsPrivateRepo() {
-		io.GitHostAccessToken = ui.EnterGitHostAccessToken(io.ServiceRepoURL)
-	}
 	io.ServiceWebhookSecret = ui.EnterServiceWebhookSecret()
-	commitStatusTrackerCheck := ui.SelectOptionCommitStatusTracker()
-	if commitStatusTrackerCheck == "yes" {
-		io.CommitStatusTracker = true
-		if io.GitHostAccessToken == "" {
-			io.GitHostAccessToken = ui.EnterGitHostAccessToken(io.ServiceRepoURL)
+	secret, err := webhook.GetAccessToken(io.ServiceRepoURL)
+	if err != nil {
+		return err
+	}
+	if err == nil && secret == "" {
+		io.GitHostAccessToken = ui.EnterGitHostAccessToken(io.ServiceRepoURL)
+		if io.GitHostAccessToken != "" {
+			err := webhook.SetSecret(io.GitOpsRepoURL, io.GitHostAccessToken)
+			if err != nil {
+				return err
+			}
+			err = webhook.SetSecret(io.ServiceRepoURL, io.GitHostAccessToken)
+			if err != nil {
+				return err
+			}
 		}
```

Comment:
Can io.GitHostAccessToken still be ""?   If so, should we ask for the user for the question for commit status tracker?

Reply:
It cannot in this case because validation will fail, prompting the user to enter the token again. If the token is not found in the keyring the user is forced to enter a valid token, else UI prompt will fail.